hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerCall.java

```diff
@@ -431,31 +436,42 @@ public boolean isClientCellBlockSupported() {
 
   @Override
   public long getResponseCellSize() {
-    return responseCellSize;
+    return responseCellSize.get();
   }
 
   @Override
-  public void incrementResponseCellSize(long cellSize) {
-    responseCellSize += cellSize;
+  public long getNumsOfGet() {
+    return getsNum.get();
+  }
+
+  public long incrementGetsNum() {
+    return this.getsNum.incrementAndGet();
   }
 
   @Override
-  public long getResponseBlockSize() {
-    return responseBlockSize;
+  public void addResultSize(Result result) {
+    if (result != null && !result.isEmpty()) {
+      for (Cell c : result.rawCells()) {
+        this.responseCellSize.addAndGet(PrivateCellUtil.estimatedSerializedSizeOf(c));
+      }
+    }
   }
 
   @Override
-  public void incrementResponseBlockSize(long blockSize) {
-    responseBlockSize += blockSize;
+  public boolean isAboveQuota(long maxQuotaResultSize, long maxRowsPerMultiGet) {
```

Comment:
This should not a method of RpcContext? This can be a static "util" method in RSRpcServices?